xapian-letor/ranker/svmranker.cc

```diff
@@ -45,6 +45,20 @@ Proceedings of the eighth ACM SIGKDD international conference on Knowledge disco
 using namespace std;
 using namespace Xapian;
 
+static void
+clear_svm_problem(svm_problem problem, int n)
```

Comment:
This passes `problem` by value, so will copy it - it's a 3 member structure, so that's not awful, but it would really make more sense to pass it by reference.

Isn't `n` redundant information?  It seems you could just use `problem.l` instead, which would make it harder to call this function incorrectly.